### ISSUE TITLE
0.16.0 corrupts package.json when un/installing a package

### ISSUE THREAD

**Issue by neothemachine** [jspm/jspm-cli#1036]
Before:
```
{
  "name": "covjson-reader",
  "version": "0.1.0",
  "main": "src/reader",
  "registry": "jspm",
  "format": "esm",
  "scripts": {
    "test": "standard"
  },
  "standard": {
    "parser": "babel-eslint",
    "ignore": [
      "src/config.js"
    ]
  },
  "jspm": {
    "directories": {
      "baseURL": "src"
    },
    "dependencies": {
      "cbor": "github:paroga/cbor-js@master"
    },
    "devDependencies": {
      "babel": "npm:babel-core@^5.8.22",
      "babel-runtime": "npm:babel-runtime@^5.8.20",
      "core-js": "npm:core-js@^1.1.0"
    }
  },
  "devDependencies": {
    "babel-eslint": "^4.0.10",
    "jspm": "^0.16.0",
    "standard": "^5.1.0"
  }
}
```

After doing `jspm install --dev mocha=github:mochajs/mocha`:
```
{
  "name": "covjson-reader",
  "version": "0.1.0",
  "main": "src/reader",
  "registry": "jspm",
  "format": "esm",
  "scripts": {
    "test": "standard"
  },
  "standard": {
    "parser": "babel-eslint",
    "ignore": [
      "src/config.js"
    ]
  },
  "jspm": {
    "directories": {
      "baseURL": "src"
    },
    "dependencies": {
      "cbor": "github:paroga/cbor-js@master"
    },
    "devDependencies": {
      "babel": "npm:babel-core@^5.8.22",
      "babel-runtime": "npm:babel-runtime@^5.8.20",
      "core-js": "npm:core-js@^1.1.0"
    }
  },
  "devDependencies": {
    "babel": "npm:babel-core@^5.8.22",
    "babel-runtime": "npm:babel-runtime@^5.8.20",
    "core-js": "npm:core-js@^1.1.0",
    "mocha": "github:mochajs/mocha@^2.2.5"
  },
  "directories": {
    "baseURL": "src"
  },
  "dependencies": {
    "cbor": "github:paroga/cbor-js@master"
  }
}
```
Note how it simply overrides the root `"devDependencies"`. jspm, standard,  babel-eslint is gone!This happens also without `--dev` and also when `uninstall`ing things.

**Comment by neothemachine** [comment]
This happens with a simple `jspm install` as well. Something is really wrong here...

**Comment by guybedford** [comment]
Are you definitely running jspm 0.16 here?

**Comment by guybedford** [comment]
I'm definitely not getting this, it could be worth double checking the version with `jspm -v`.

**Comment by neothemachine** [comment]
Yep:
"0.16.0
Running against local jspm install."

**Comment by guybedford** [comment]
Thanks, fixed in https://github.com/jspm/jspm-cli/commit/fd4eff1a891ce0bf7c08acf97ac8524388ea5472.

**Comment by neothemachine** [comment]
Thanks, but just to clarify, the way I wrote/use the package.json is 
idiomatic right? When I put all the jspm deps in the root then I cannot 
use npm anymore for installing (and running) dev dependencies, right?

**Comment by guybedford** [comment]
The issue was that the `registry` property was at the root, while the other properties were prefixed, which made jspm think all the properties were at the root. What you're doing seems fine to do though.

**Comment by guybedford** [comment]
Released in 0.16.1.

**Comment by neothemachine** [comment]
Works, thanks!
